modin/engines/dask/pandas_on_dask/frame/partition.py

```diff
@@ -143,16 +129,29 @@ def add_to_apply_calls(self, func, **kwargs):
         The keyword arguments are sent as a dictionary.
         """
         return PandasOnDaskFramePartition(
-            self.future, call_queue=self.call_queue + [[pkl.dumps(func), kwargs]]
+            self.future, call_queue=self.call_queue + [[func, args, kwargs]]
         )
 
     def drain_call_queue(self):
         """Execute all operations stored in the call queue on the object wrapped by this partition."""
         if len(self.call_queue) == 0:
             return
-        new_partition = self.apply(lambda x: x)
-        self.future = new_partition.future
-        self._ip_cache = new_partition._ip_cache
+        call_queue = self.call_queue
+        client = default_client()
+        if len(call_queue) > 1:
+            future = client.submit(
+                apply_list_of_funcs, self.future, call_queue, pure=False
+            )
+        else:
+            # We handle `len(call_queue) == 1` in a different way because
+            # this improves performance a bit.
+            func, args, kwargs = call_queue[0]
+            future = client.submit(apply_func, self.future, func, *args, **kwargs)
+        futures = [
+            client.submit(lambda l, i: l[i], future, i, pure=False) for i in range(2)
+        ]
```

Comment:
can we extract this in a common function? I think this piece of code is exactly identical in `apply()` and here...

Reply:
I think this relates to this conversation ([link](https://github.com/modin-project/modin/pull/2919#discussion_r652781272)) and it makes sense to be fixed later.

Reply:
Makes sense, let's fix all at once in a later PR.